parsons/databases/redshift/redshift.py

```diff
@@ -785,10 +797,20 @@ def upsert(self, table_obj, target_table, primary_key, vacuum=True, distinct_che
                     # column is not impactful barely impactful
                     # https://docs.aws.amazon.com/redshift/latest/dg/c_Loading_tables_auto_compress.html
                     copy_args = dict(copy_args, compupdate=False)
-                self.copy(table_obj, staging_tbl,
-                          template_table=target_table,
-                          alter_table=False,  # We just did our own alter table above
-                          **copy_args)
+                if from_s3:
+                    if table_obj is not None:
+                        raise ValueError(
+                            'upsert() using from_s3= requires the first argument (table_obj)'
```

Comment:
Did you mean to have `using from_s3=True` here?

Reply:
I didn't, but I'm happy to make the error text whatever you think is best (and that's a good suggestion!)

Reply:
Honestly, entirely up to you. Feel free to make the change, or merge as you see fit!

Reply:
Ok, I tweaked it.  I don't think I have power to merge, but I'm happy if you are :pray: